sonar-ruby-plugin/src/main/java/org/sonarsource/ruby/converter/RubyVisitor.java

```diff
@@ -91,11 +94,23 @@
     UNARY_OPERATOR_MAP.put("not", UnaryExpressionTree.Operator.NEGATE);
   }
 
+  private Deque<String> nodeTypeStack = new ArrayDeque<>();
+
   public RubyVisitor(TreeMetaDataProvider metaDataProvider) {
     this.metaDataProvider = metaDataProvider;
   }
 
-  public Tree visitNode(AstNode node, List<Object> children) {
+  public void beforeVisit(AstNode node) {
+    nodeTypeStack.push(node.type());
+  }
+
+  public Tree afterVisit(AstNode node, List<Object> children) {
```

Comment:
I find it strange to have the `afterVisit` method calling `visitNode`.
I'd prefer to have all 3 methods called individually by the ruby processor one after the other